tests/type_inference/test_boolops.py

```diff
@@ -0,0 +1,34 @@
+import astroid
+import nose
+from hypothesis import given, settings, assume
+import hypothesis.strategies as hs
+import tests.custom_hypothesis_support as cs
+from typing import Any
+settings.load_profile("pyta")
+
+
+@given(hs.lists(cs.binary_bool_operator), hs.lists(cs.primitive_values))
```

Comment:
Can you mix `and` and `or` in a single `BoolOp` node? If not, what you should do take in a single bool op, and a list of values (of length `>= 2`) and then join the list using the single operation.

You should also have two different cases, one with a homogeneous list, and one with a heterogeneous list.